### DISCUSSION TITLE
Behaviour of a unconstrained variable

### DISCUSSION THREAD

**Discussion by tcinbis** [google/or-tools#2599] [CP-SAT questions] [answered]
Hey all,

My question is how the CpSolver treats, for example, boolean variables, which are added to the model but never used in any constraint. In a sense, they are free or unconstrained. My current understanding and intuition are that they should take any value possible to produce a feasible solution (assuming other constraints in a model do not make the problem infeasible).

OR

Does it take some fixed default value? 
Except for runtime, could they affect the solution?

I really appreciate any help!

**Comment by lperron** [comment, marked as answer]
Presolve will remove them from the internal model and assign them a default value.

**Comment by gregy4** [comment]
During presolve phase is removed a variable for which exists a value that doesn't negatively affect feasibility of a model. For unconstrained variable it is default value, in other cases it depends on a constraint to be true.

**Comment by lperron** [comment]
I see someone has read the presolve code :-)
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00
